src/main/java/org/cactoos/io/LoggingInputStream.java

```diff
@@ -0,0 +1,236 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Logged input stream.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.29
+ */
+@SuppressWarnings("PMD.LoggerIsNotStaticFinal")
+public final class LoggingInputStream extends InputStream {
+
+    /**
+     * The input stream.
+     */
+    private final InputStream origin;
+
+    /**
+     * Where the data comes from.
+     */
+    private final String source;
+
+    /**
+     * The logger.
+     */
+    private final Logger logger;
+
+    /**
+     * The bytes read.
+     */
+    private final AtomicLong bytes;
```

Comment:
@fabriciofx If the class does not offer thread safety, why use Atomic? Is it thread safe as it is now? Then update the Javadoc. And if it's not thread safe anyway, I would use Long, for simplicity... or you have another reason for Atomic?

Reply:
@amihaiemil TThis class needs maintain the state (sum of all bytes). So, it class isn't thread safe, but I used the same solution used by Yegor in [IteratorNoNulls](https://github.com/yegor256/cactoos/blob/master/src/main/java/org/cactoos/iterator/IteratorNoNulls.java).

Reply:
@fabriciofx ok, let's leave it like that then